fulp_modules/lisa/joel_gun.dm

```diff
@@ -0,0 +1,133 @@
+	if(used_ability)
+		to_chat(user, "<span class='warning'>You have to wait before using an ability!</span>")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(user == victim)
+		to_chat(user,"<span class='warning'>You can't steady aim yourself!</span>")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	user.visible_message("<span class='danger'>[user.name] holds [victim.name] at gunpoint!</span>",\
+	 "<span class='userdanger'>You hold [victim.name] up at gunpoint!</span>",\
+	 "<span class='hear'>You hear metal clanking...</span>")
+	playsound(loc, 'fulp_modules/lisa/Sounds/steadyaim.ogg', 50, FALSE, -5)
+	if(prob(50))
+		victim.Stun(rand(10, 20))
+	used_ability = TRUE
+	addtimer(CALLBACK(src, .proc/clear_cooldown), 5 SECONDS)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/// Prevents the gun from being fired.
+/obj/item/gun/ballistic/revolver/joel/afterattack(atom/target, mob/living/user, flag, params)
+	if(!chambered) /// Can shoot if its empty, we don't care!
+		user.visible_message("<span class='danger'>*click*</span>")
+		playsound(loc, 'fulp_modules/lisa/Sounds/gunclick.ogg', 50, FALSE, -5)
+		return
+	if(used_ability)
+		to_chat(user, "<span class='warning'>You don't want to waste [src]'s only bullet!</span>")
+		return
+	if(target) /// Misdirection (Scholar of the Wilbur Sin exclusive)
+		user.visible_message("<span class='danger'>[user.name] points their gun towards [target]!</span>",\
+		 "<span class='userdanger'>You misdirect your gun towards [target]!</span>",\
+		 "<span class='hear'>You hear a gasp...</span>")
+		user.emote("gasp")
+		playsound(loc, 'fulp_modules/lisa/Sounds/misdirection.ogg', 50, FALSE, -5)
+		used_ability = TRUE
+		addtimer(CALLBACK(src, .proc/clear_cooldown), 5 SECONDS)
+		for(var/mob/living/H in view(1, target))
+			if(user.body_position == STANDING_UP)
+				user.visible_message("<span class='danger'>You quickly jump towards [target]!</span>",\
+				 "<span class='userdanger'>[H] quickly jumps towards [target]!</span>",\
+				 "<span class='hear'>You hear aggressive shuffling!</span>")
+				H.Move(target)
+		for(var/mob/living/L in viewers(5, target))
+			L.face_atom(target)
+			L.do_alert_animation()
+		return
+
+/*
+ *	Below is the cylinder, bullet and casing.
+ *	We're using unique ones so people cant get more bullets for anything, yada yada, whatever.
+ *	Unique gun, unique single bullet. You lose it, it's gone.
+ */
+
+/// Cylinder
+/obj/item/ammo_box/magazine/internal/cylinder/c22
+	name = "\improper Bolt Action revolver cylinder"
+	ammo_type = /obj/item/ammo_casing/c22
+	caliber = CALIBER_C22
+	max_ammo = 1
+	multiload = FALSE
+
+/// Bullet casing & Bullet
+/obj/item/ammo_casing/c22
+	name = "5.6mm bullet casing"
+	desc = "An incredibly rare type of bullet."
+	caliber = CALIBER_C22
+	projectile_type = /obj/projectile/bullet/c22
+
+/// This isnt meant to be fired, so do 0 damage. This can technically be fired by dual wielding, so they'll have a nice surprise of a big load of nothing if they do.
```

Comment:
you can fix the dual-wielding issue by just overriding process_fire() like russian revolvers do

Reply:
I'll see once I get around to re-doing this PR, I got a few things I want to change but we're currently testmerging the TGU